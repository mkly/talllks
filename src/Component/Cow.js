import { html } from "htm/preact";
import { css } from "emotion";

const containerStyle = css`
  width: 95%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const innerContainerStyle = css`
  text-align: center;
  overflow: hidden;
`;

const cowStyle = css`
  display: inline-block;
  text-align: left;
`;

import { say } from "cowsay";

const Cow = ({ cow }) => html`
  <div class=${containerStyle}>
    <div class=${innerContainerStyle}>
      <pre class=${cowStyle}>
${say({ text: "Hello, CircleCI!", cow, mode: "r" })}</pre
      >
    </div>
  </div>
`;

export default Cow;
