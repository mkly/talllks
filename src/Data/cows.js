import {
  DEFAULT,
  BEAVIS_ZEN,
  BUD_FROGS,
  BUNNY,
  CHEESE,
  COWER,
  DAEMON,
  DOGE,
  DRAGON_AND_COW,
  DRAGON,
  ELEPHANT_IN_SNAKE,
  EYES,
  FLAMING_SHEEP,
  GHOSTBUSTERS,
  GOAT,
  HEDGEHOG,
  HELLOKITTY,
  KITTY,
  KOALA,
  KOSH,
  LUKE_KOALA,
  MECH_AND_COW,
  MEOW,
  MILK,
  MOOFASA,
  MOOSE,
  MUTILATED,
  REN,
  SHEEP,
  SKELETON,
  SMALL,
  SQUIRREL,
  STEGOSAURUS,
  STIMPY,
  SURGERY,
  TURKEY,
  TURTLE,
  TUX,
  VADER_KOALA,
  WHALE,
  WWW
} from "cowsay";

export default [
  DEFAULT,
  BEAVIS_ZEN,
  BUD_FROGS,
  BUNNY,
  CHEESE,
  COWER,
  DAEMON,
  DOGE,
  DRAGON_AND_COW,
  DRAGON,
  ELEPHANT_IN_SNAKE,
  EYES,
  FLAMING_SHEEP,
  GHOSTBUSTERS,
  GOAT,
  HEDGEHOG,
  HELLOKITTY,
  KITTY,
  KOALA,
  KOSH,
  LUKE_KOALA,
  MECH_AND_COW,
  MEOW,
  MILK,
  MOOFASA,
  MOOSE,
  MUTILATED,
  REN,
  SHEEP,
  SKELETON,
  SMALL,
  SQUIRREL,
  STEGOSAURUS,
  STIMPY,
  SURGERY,
  TURKEY,
  TURTLE,
  TUX,
  VADER_KOALA,
  WHALE,
  WWW
];
