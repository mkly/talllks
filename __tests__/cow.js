/**
 * @jest-environment jsdom
 */
import { html, render } from "htm/preact";

import cows from "../src/Data/cows";
import Cow from "../src/Component/Cow";

describe("Cow", () => {
  it("Should show default cow", () => {
    const body = document.createElement("div");
    document.body.appendChild(body);

    render(html`<${Cow}/>`, body);
    expect(body.innerHTML).toMatchSnapshot();
  });
});
