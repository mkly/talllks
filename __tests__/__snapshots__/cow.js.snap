// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cow Should show default cow 1`] = `
"<div class=\\"css-1wi0k2z\\"><div class=\\"css-om2kyi\\"><pre class=\\"css-2gq9ut\\"> __________________
&lt; Hello, CircleCI! &gt;
 ------------------
        \\\\   ^__^
         \\\\  (oo)\\\\_______
            (__)\\\\       )\\\\/\\\\
                ||----w |
                ||     ||</pre></div></div>"
`;
